tests/integration/interface_common_test.py

```diff
@@ -34,3 +39,32 @@ def test_iface_description_removal(eth1_up):
     libnmstate.apply(desired_state)
     current_state = statelib.show_only(('eth1',))
     assert Interface.DESCRIPTION not in current_state[Interface.KEY][0]
+
+
+def test_take_over_virtual_interface_then_remove():
```

Comment:
Could you please add a second test that does the same except triggering a rollback instead of a successful apply to check that the device is restored by the rollback?

Reply:
We already cover these scenarios using bond and bridge rollback.

Reply:
Do the bond and bridge rollback tests really test that an interface that was not created by NM is restored on failure?

Reply:
Nope, that is not tested. Good point, thanks.

Reply:
Done, thanks